contracts/exchangeIssuance/IssueRedeem.sol

```diff
@@ -0,0 +1,192 @@
+pragma solidity >=0.6.10;
+pragma experimental ABIEncoderV2;
+
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+import "@uniswap/v2-periphery/contracts/interfaces/IUniswapV2Router02.sol";
+import "@uniswap/v2-core/contracts/interfaces/IUniswapV2Factory.sol";
+import "../lib/UniswapV2Library.sol";
+import "../lib/SushiswapV2Library.sol";
+import "../interfaces/ISetToken.sol";
+import "../interfaces/IBasicIssuanceModule.sol";
+import "../interfaces/IWETH.sol";
+
+/**
+ * @title IssueRedeem
+ * @author Noah Citron
+ *
+ * Contract that allows for the minting and redeeming any Set token using
+ * ETH as the paying/receiving currency. All swaps are done using the best price
+ * found on Uniswap or Sushiswap.
+ *
+ */
+contract IssueRedeem {
+
+    using SafeMath for uint256;
+    
+    /* ============ State Variables ============ */
+
+    IUniswapV2Router02 private uniRouter;
+    address private uniFactory;
+    IUniswapV2Router02 private sushiRouter;
+    address private sushiFactory;
+
+    IBasicIssuanceModule private basicIssuanceModule;
+    address constant private WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    uint256 constant private MAX_INT = 2**256 - 1;
+
+    /* ============ Constructor ============ */
+
+    constructor() public {
```

Comment:
Instead of hardcoding values, we typically prefer to pass in addresses via the constructor so that we are able to test the contract in different environments (kovan, main-net, etc.)